Varlink/varlink_service_test.py

```diff
@@ -0,0 +1,110 @@
+import pytest
+import libnmstate
+import varlink
+import threading
+
+from nmstate_varlink import get_varlink_server
+from libnmstate.schema import Interface
+from libnmstate.schema import InterfaceType
+from libnmstate.schema import InterfaceState
+from libnmstate.schema import InterfaceIPv4
+from libnmstate.schema import InterfaceIPv6
+from libnmstate.schema import DNS
+from libnmstate.schema import Route
+from libnmstate.schema import RouteRule
+
+
+VARLINK_INTERFACE = "io.nmstate"
+
+
+@pytest.fixture(
+    params=["unix:/tmp/nmstate", "unix:@nmstate", "tcp:127.0.0.1:8000"],
+    ids=[
+        "unix_socket_connection",
+        "unix_namespaced_socket_connection",
+        "tcp_connection",
+    ],
+)
+def server(request):
+    server = get_varlink_server(request.param)
+    thread_server = threading.Thread(target=server.serve_forever, daemon=True)
+    thread_server.start()
+    return server
+
+
+def _format_address(server_address):
+    if isinstance(server_address, tuple):
+        return "tcp:" + str(server_address[0]) + ":" + str(server_address[-1])
+    else:
+        return "unix:" + str(server_address)
+
+
+def test_varlink_show(server):
+    lib_state = libnmstate.show()
+    with varlink.Client(_format_address(server.server_address)).open(
+        VARLINK_INTERFACE, namespaced=False
+    ) as con:
+        varlink_state = con._call("Show")
+        assert lib_state[DNS.KEY] == varlink_state["state"][DNS.KEY]
```

Comment:
why don't you assert `varlink_["state"] == `lib_state` but compare the subtrees? Also, in my opinion the expected value belongs on the right-hand side. so you can read it as "assert that the varlink state is the library state".